library/manageiq_provider.py

```diff
@@ -577,8 +598,10 @@ def main():
     if state == 'present':
         if provider_type in ("openshift-enterprise", "openshift-origin"):
             endpoints = [manageiq.generate_auth_key_config('default', 'bearer', hostname, port, token, provider_verify_ssl, provider_ca_path)]
-            if module.params['metrics']:
-                endpoints.append(manageiq.generate_auth_key_config('hawkular', 'hawkular', h_hostname, h_port, token, provider_verify_ssl, provider_ca_path))
+            if monitoring:
+                monitoring_hostname =  module.params['monitoring_hostname']
+                monitoring_port = module.params['monitoring_port']
+                endpoints.append(manageiq.generate_auth_key_config(monitoring, monitoring, monitoring_hostname, monitoring_port, token, provider_verify_ssl, provider_ca_path))
```

Comment:
Your call ;-).
Personally I find I can't read the current `generate_auth_key_config()` call without going to see its definition.  I suppose if you change this, should also change the 2 other endpoint-generating calls.